### PR TITLE
Added support for constant term in the linear dynamic constraints

### PR DESCRIPTION
**Pull request by dlcole3**
For $x_{t + 1} = Ax_t + bu_t + w$, The vector `w` changes the constraint bounds and the linear/constant terms of the objective function. I derived the equations for these and updated the `SparseLQDynamicModel` and `DenseLQDynamicModel` to operate with `w`. I added tests to check this, and I updated the JuMP model that we used for comparison. This also required updating the `DenseLQDynamicBlocks` significantly to store the needed data to reset `s0`. 

For reference, the equation used for deriving this is $\boldsymbol{x} = \boldsymbol{A} \hat{x} + \boldsymbol{B} \boldsymbol{v} + \tilde{\boldsymbol{A}}\boldsymbol{w}$. In this case, $\tilde{\boldsymbol{A}}$ is the same format as $\boldsymbol{B}$ but with the identity matrix in place of the $B$ matrix.

### CODE REVIEW

**Comment by sshin23** [comment]
@dlcole3 can't `w` be time-varying? When we have `w`, typically we have time-variability in `w`

File: src/LinearQuadratic/LinearQuadratic.jl [hunk 1]
```diff
@@ -192,35 +199,64 @@ struct SparseLQDynamicModel{T, V, M1, M2, M3, MK} <:  AbstractDynamicModel{T,V}
...
 
 See also `reset_s0!`
 """
-struct DenseLQDynamicBlocks{T, M}
-    # block_h0 = A^T((Q + KTRK + 2 * SK))A where Q, K, R, S, and A are block matrices
-    # block_h  = (QB + SKB + K^T R K B + K^T S^T B)^T A + (S + K^T R)^T A
-    # block_d  = (E + FK) A
-    # block_KA = KA
+mutable struct DenseLQDynamicBlocks{T, V, M}
+    # Aw = block_matrix_A * w (result is a Vector; block_matrix A is like block_B,
```
Comment (frapac) [inline src/LinearQuadratic/LinearQuadratic.jl]: Can we move the comments in front of the attribute in the struct?

**Review by frapac — APPROVED** [review]
it overall looks good to me!

**Comment by dlcole3** [comment]
> @dlcole3 can't `w` be time-varying? When we have `w`, typically we have time-variability in `w`

@sshin23 It could be time varying, and that would be a relatively simple change to make. I will update that. In the case of the application in the second example problem (oscillating masses), they are not time varying.

**Review by sshin23 — APPROVED** [review]
If time-varying is taking long time, feel free to merge this first and open an issue.

The change looks good

**Review by sshin23 — APPROVED** [review]

**Pull request merged**